Common/src/dual_grid_structure.cpp

```diff
@@ -424,6 +446,27 @@ void CPoint::SetBoundary(unsigned short val_nmarker) {
 
 }
 
+void CPoint::Set_AdjIndices(bool input) {
+  for (unsigned short iDim = 0; iDim < nDim; iDim++) {
+    if(input) {
+      AD::Set_AdjIndex(Input_AdjIndices[iDim], Coord[iDim]);
+    }
+    else {
+      AD::Set_AdjIndex(Output_AdjIndices[iDim], Coord[iDim]);
+    }
+  }
+}
+
+void CPoint::SetAdjointSolution(su2double *adj_sol) {
+  for (unsigned short iDim = 0; iDim < nDim; iDim++) {
+    AD::SetDerivative(Output_AdjIndices[iDim], SU2_TYPE::GetValue(adj_sol[iDim]));
+  }
+}
+
+void CPoint::GetAdjointSolution(su2double *adj_sol, unsigned short iDim) {
```

Comment:
please return by value, this looks weird (and what happens if people pass NULL?)

Reply:
No idea how that happened. Thanks